File: algorithms/medium/minimize_maximum_pair_sum_in_array.rs
impl Solution {
    pub fn min_pair_sum(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut ans = 0;
        let mut i = 0;
        let mut j = nums.len() - 1;
        while i < j {
            ans = ans.max(nums[i] + nums[j]);
            i += 1;
            j -= 1;
        }
        ans
    }
}

File: algorithms/medium/minimize_maximum_pair_sum_in_array.kt
class Solution {
  fun minPairSum(nums: IntArray): Int {
    nums.sort()
    var i = 0
    var j = nums.size - 1
    var ans = 0
    while (i < j) {
      ans = max(ans, nums[i] + nums[j])
      i += 1
      j -= 1
    }
    return ans
  }
}
